ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -174,7 +172,20 @@ impl<'a> X64CodeGen<'a> {
         inst: &jit_ir::Instruction,
     ) -> Result<(), CompilationError> {
         #[cfg(any(debug_assertions, test))]
-        self.comment(self.asm.offset(), inst.to_string(self.jit_mod).unwrap());
+        {
+            if let Some(dt) = inst.def_type(self.jit_mod) {
```

Comment:
This feels like it's in the wrong place. Shouldn't this be encapsulated inside a struct in the JIT IR?

Reply:
Good question! The basic answer is because an `Instruction` doesn't know its index in the trace. The previous `LocalNumbers` struct was one way around this, but quite complex.

I suppose one way around this would be a method on (probably) `Instruction` called something like `to_string_with_idx` that would do the `%0 = ` bit?

Reply:
Hrm. I don't really want to have to think about whether to call `to_string_with_idx` or `to_string`.

Reply:
We *could* I suppose do the old "lookup the instruction index in the `Module`" trick, but it feels really fragile to me. It took me ages to follow through the code to realise how it worked. But I don't feel strongly about it.

Reply:
Let me quickly see if I can knock up a prototype.

Reply:
Before you spend time on this, check the comment below about the overall approach.